io_gripper_controller/src/io_gripper_controller.cpp

```diff
@@ -0,0 +1,1376 @@
+    case gripper_state_type::CLOSE_GRIPPER:
+      for (size_t i = 0; i < command_ios_close.size(); ++i)
+      {
+        setResult = find_and_set_command(command_ios_close[i], command_ios_close_values[i]);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(
+            get_node()->get_logger(), "Failed to set the command state for %s",
+            command_ios_close[i].c_str());
+        }
+      }
+
+      gripper_state_buffer_.writeFromNonRT(gripper_state_type::CHECK_GRIPPER_STATE);
+      break;
+    case gripper_state_type::CHECK_GRIPPER_STATE:
+      for (const auto & [state_name, state_params] : params_.close.state.possible_closed_states_map)
+      {
+        check_state_ios_ = false;
+        for (const auto & high_val : state_params.high)
+        {
+          setResult = find_and_get_state(high_val, state_value_);
+          if (!setResult)
+          {
+            RCLCPP_ERROR(
+              get_node()->get_logger(), "Failed to get the state for %s", high_val.c_str());
+          }
+          else
+          {
+            if (abs(state_value_ - 1.0) < std::numeric_limits<double>::epsilon())
+            {
+              check_state_ios_ = true;
+            }
+            else
+            {
+              check_state_ios_ = false;
+              break;
+            }
+          }
+        }
+        for (const auto & low_val : state_params.low)
+        {
+          setResult = find_and_get_state(low_val, state_value_);
+          if (!setResult)
+          {
+            RCLCPP_ERROR(
+              get_node()->get_logger(), "Failed to get the state for %s", low_val.c_str());
+          }
+          else
+          {
+            if (abs(state_value_ - 0.0) < std::numeric_limits<double>::epsilon())
+            {
+              check_state_ios_ = true;
+            }
+            else
+            {
+              check_state_ios_ = false;
+              break;
+            }
+          }
+        }
+        if (check_state_ios_)
+        {
+          closed_state_name_ = state_name;
+          gripper_state_buffer_.writeFromNonRT(gripper_state_type::SET_AFTER_COMMAND);
+          break;
+        }
+      }
+      break;
+    case gripper_state_type::SET_AFTER_COMMAND:
+      closed_state_values_ = params_.close.state.possible_closed_states_map.at(closed_state_name_);
+
+      for (const auto & high_val : closed_state_values_.set_after_command_high)
+      {
+        setResult = find_and_set_command(high_val, 1.0);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(
+            get_node()->get_logger(), "Failed to set the command state for %s", high_val.c_str());
+        }
+      }
+
+      for (const auto & low_val : closed_state_values_.set_after_command_low)
+      {
+        setResult = find_and_set_command(low_val, 0.0);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(
+            get_node()->get_logger(), "Failed to set the command state for %s", low_val.c_str());
+        }
+      }
```

Comment:
This code has two parts that are 98% identical. This should go into a function. Also this should be updated for all before/after, or if even possible setting commands in both function. We can reduce at least 50 lines of code doing that.